BonusCalcListener/UseCase/PayElementMapper.cs

```diff
@@ -0,0 +1,38 @@
+using BonusCalcListener.Boundary;
+using BonusCalcListener.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BonusCalcListener.UseCase
+{
+    public class PayElementMapper : IMapPayElements
+    {
+        const int REACTIVE_REPAIRS_TYPE = 301;
+        public PayElement BuildPayElement(WorkOrderOperativeSmvData eventData, Timesheet operativeTimesheet)
+        {
+            if (!eventData.ClosedTime.HasValue) throw new ArgumentException($"Cannot create a pay element with null closed time, WorkOrder: {eventData.WorkOrderId}, operative: {eventData.OperativeId}");
+
+            if (!eventData.JobPercentage.HasValue || eventData.JobPercentage <= 0) throw new ArgumentException($"Cannot have a job percentage of zero or less than zero, WorkOrder: {eventData.WorkOrderId}");
+
+            var operativeJobSmvHours = (decimal) (eventData.StandardMinuteValue * eventData.JobPercentage ?? 0) / 60;
+
+            return new PayElement
+            {
+                TimesheetId = operativeTimesheet.Id,
+                PayElementTypeId = REACTIVE_REPAIRS_TYPE,
+                Address = eventData.Address,
+                WorkOrder = eventData.WorkOrderId,
+                Duration = operativeJobSmvHours,
+                ReadOnly = true,
+                Monday = eventData.ClosedTime?.DayOfWeek == DayOfWeek.Monday ? operativeJobSmvHours : 0.0m,
```

Comment:
For a job that was completed in a week that has been closed should the time go to the Monday of the earliest week that's still open or the day of the week that the job was closed on? Not sure it matters either way.

Reply:
The advantage of doing it on the day of the week is that, if it's just two weeks overlaid on top each other it is still _almost_ decodable by a savvy operative. 14 could reasonably be assumed to be 7 + 7. If the whole week goes on to Monday, Monday becomes an intelligible blob in terms of reconciliation. The disadvantage is that it might make you think you've done more work than you actually have. I'll leave it as is for now.